### PR TITLE
Address Findbugs and PMD warnings

### PR DESCRIPTION
**Pull request by chanseokoh**
#241 (FindBugs and PMD)

- FindBugs: use StringBuilder in a loop for string concat (2 instances)
- PMD: mostly about combining nested if-statements and ignoring empty catch-blocks, plus some minor things

### CODE REVIEW

File: common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java [hunk 1]
```diff
@@ -153,7 +153,7 @@ public static void assertIsSerializable(@NotNull Serializable object) {
         try {
           out.close();
         } catch (IOException e) {
-          // ignore
+          fail("Cannot close output stream: " + e.getMessage());
```
Comment (elharo) [inline common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java]: I don't think we do want to fail here.

**Comment by patflynn** [comment]
I'd like to leave the breaking up of conditionals into nests up to the discretion of the author as much as possible. Sometimes it's more readable to nest.. In these particular cases I'm not strongly opionionated.

**Comment by patflynn** [comment]
LGTM once elharo's comment is addressed.

File: google-account-plugin/src/com/google/gct/stats/GoogleUsageTracker.java [hunk 2]
```diff
@@ -71,27 +71,26 @@ public void trackEvent(@NotNull String eventCategory,
...
-            if (UsageTrackerManager.getInstance().isTrackingEnabled()) {
-                ArrayList postData = Lists.newArrayList(analyticsBaseData);
-                postData.add(new BasicNameValuePair("tid", analyticsId));
-                postData.add(new BasicNameValuePair(INTELLIJ_EDITION, PlatformUtils.isCommunityEdition() ? "community" : "ultimate"));
-                postData.add(new BasicNameValuePair("ec", eventCategory));
-                postData.add(new BasicNameValuePair("ea", eventAction));
-                if (!Strings.isNullOrEmpty(eventLabel)) {
-                    postData.add(new BasicNameValuePair("el", eventLabel));
-                }
+        if ((!ApplicationManager.getApplication().isUnitTestMode()) &&
```
Comment (elharo) [inline google-account-plugin/src/com/google/gct/stats/GoogleUsageTracker.java]: I'm not sure this one pulls its weight. It seems marginally harder to read. Maybe of you removed the extra parentheses?

**Comment by patflynn** [comment]
(not that I have an opinion)

File: google-cloud-tools-plugin/src/com/google/gct/idea/ui/CustomizableComboBox.java [hunk 3]
```diff
@@ -190,10 +190,9 @@ private static boolean isUsingDarculaUIFlavor() {
   @Override
   public void paint(Graphics g) {
     super.paint(g);
-    if (textField.isFocusOwner() || (getPopup() != null && getPopup().isPopupVisible())) {
-      if (isUsingDarculaUIFlavor()) {
-        DarculaUIUtil.paintFocusRing(g, 3, 3, getWidth() - 4, getHeight() - 4);
-      }
+    if ((textField.isFocusOwner() || (getPopup() != null && getPopup().isPopupVisible())) &&
```
Comment (elharo) [inline google-cloud-tools-plugin/src/com/google/gct/idea/ui/CustomizableComboBox.java]: Ditto. IMHO this was easier to read before.

**Comment by chanseokoh** [comment]
I agree about the nested if's. I'll revert most of the changes back (this would introduce // NOPMD comments). I'll combine only if they are very short.

File: common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java (hunk 1 above)
Comment (chanseokoh) [inline common-test-lib/src/main/java/com/google/gct/idea/testing/TestUtils.java]: Reverted back.

**Comment by chanseokoh** [comment]
Reverted most of the combined if's. I'll merge it soon.

**Pull request merged**